### PR TITLE
Merge course type into develop

### PR DESCRIPTION
**Pull request by karmetun**

### CODE REVIEW

File: src/main/java/ProjectOcean/Model/Course.java [hunk 1]
```diff
@@ -54,6 +54,15 @@ public String getCourseDescription() {
         return cDetails.getRequiredCourses();
     }
 
+    /**
+     *
```
Comment (Tetre2) [inline src/main/java/ProjectOcean/Model/Course.java]: remove emptyline

File: src/main/java/ProjectOcean/Model/CourseDetails.java [hunk 2]
```diff
@@ -47,6 +49,14 @@ public String getCourseDescription() {
         return courseDescription;
     }
 
+    /**
+     *
```
Comment (Tetre2) [inline src/main/java/ProjectOcean/Model/CourseDetails.java]: remove emptyline

File: src/main/java/ProjectOcean/Model/ICourse.java [hunk 3]
```diff
@@ -24,5 +24,7 @@
 
     String getCourseDescription();
 
+    String getCourseTypes();
```
Comment (Tetre2) [inline src/main/java/ProjectOcean/Model/ICourse.java]: add javaDoc to this and to the above methods in the interface

File: src/main/java/ProjectOcean/Controller/CourseListIconController.java [hunk 4]
```diff
@@ -56,6 +60,20 @@ private void populateIcon() {
...
+        String courseTypes = course.getCourseTypes();
+        if(courseTypes.contains("IT")) {
+            this.itIndicator.setStyle("-fx-background-color: #40E0D0");
+        }
+        if(courseTypes.contains("MA")) {
+            this.mathIndicator.setStyle("-fx-background-color: #DF3C3C");
+        }
+        if(courseTypes.contains("NA")) {
+            this.natureIndicator.setStyle("-fx-background-color: #3DC134");
+        }
```
Comment (Tetre2) [inline src/main/java/ProjectOcean/Controller/CourseListIconController.java]: Is there any better way of doing this?

File: src/main/java/ProjectOcean/Controller/DetailedController.java [hunk 5]
```diff
@@ -103,6 +106,32 @@ private void setLanguage(String language) {
...
+            }
+        }
+        if(courseTypes.contains("NA")) {
+            if(formatedCourseTypes.isEmpty()) {
+                formatedCourseTypes = "Naturvetenskap";
+            } else {
+                formatedCourseTypes = formatedCourseTypes + ", naturvetenskap";
+            }
+        }
+        return formatedCourseTypes;
```
Comment (Tetre2) [inline src/main/java/ProjectOcean/Controller/DetailedController.java]: Is there any better way of doing this?

File: src/main/java/ProjectOcean/Model/Course.java [hunk 6]
```diff
@@ -54,6 +54,15 @@ public String getCourseDescription() {
...
+     * @return this course's course types
+     */
+    @Override
+    public String getCourseTypes() {
+        return cDetails.getCourseTypes();
+    }
```
Comment (Tetre2) [inline src/main/java/ProjectOcean/Model/Course.java]: There is no test for this method

**Review by Tetre2 — CHANGES_REQUESTED** [review]
11 tests are failing but thay shuld sort them selves out when you merge with develop. Do that and check if there are any tests that still fails.

File: src/main/java/ProjectOcean/Controller/CourseListIconController.java (hunk 4 above)
Comment (karmetun) [reply src/main/java/ProjectOcean/Controller/CourseListIconController.java]: Fixed it

File: src/main/java/ProjectOcean/Controller/DetailedController.java (hunk 5 above)
Comment (karmetun) [reply src/main/java/ProjectOcean/Controller/DetailedController.java]: Yes,  is better now

File: src/main/java/ProjectOcean/Controller/CourseListIconController.java [hunk 7]
```diff
@@ -64,21 +68,34 @@ private void populateIcon() {
...
+            }
         }
     }
 
+    private void paintIndicator(String colour) {
+        Pane sTypeIndicator = new Pane();
+        sTypeIndicator.setStyle("-fx-background-color: " + colour);
+        sTypeIndicator.setPrefHeight(200);
+        typeIndicator.getChildren().add(sTypeIndicator);
+    }
```
Comment (Tetre2) [inline src/main/java/ProjectOcean/Controller/CourseListIconController.java]: what does "s" in sTypeIndecator mean?
maybe ues a better name

File: src/main/java/ProjectOcean/Controller/DetailedController.java [hunk 8]
```diff
@@ -106,30 +105,8 @@ private void setLanguage(String language) {
...
-        if(courseTypes.contains("NA")) {
-            if(formatedCourseTypes.isEmpty()) {
-                formatedCourseTypes = "Naturvetenskap";
-            } else {
-                formatedCourseTypes = formatedCourseTypes + ", naturvetenskap";
-            }
-        }
-        return formatedCourseTypes;
+    private void setCourseType(List<String> courseTypes) {
+        this.courseTypeLabel.setText(courseTypes.toString());
```
Comment (Tetre2) [inline src/main/java/ProjectOcean/Controller/DetailedController.java]: Can you trimm the [] of the output

File: src/main/java/ProjectOcean/Controller/RemoveMovableChild.java [hunk 9]
```diff
@@ -0,0 +1,6 @@
+package ProjectOcean.Controller;
+
+@FunctionalInterface
+public interface RemoveMovableChild {
+    void removeMovableChild(Movable movable);
```
Comment (Tetre2) [inline src/main/java/ProjectOcean/Controller/RemoveMovableChild.java]: javadoc

**Review by Tetre2 — CHANGES_REQUESTED** [review]

File: src/main/java/ProjectOcean/Controller/RemoveMovableChild.java (hunk 9 above)
Comment (karmetun) [reply src/main/java/ProjectOcean/Controller/RemoveMovableChild.java]: Not my interface, so let someone else do this, will write it onto refactor doc

**Review by Tetre2 — APPROVED** [review]

**Pull request merged**